manager/orchestrator/restart/restart.go

```diff
@@ -214,8 +219,8 @@ func (r *Supervisor) shouldRestart(ctx context.Context, t *api.Task, service *ap
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	restartInfo := r.history[instanceTuple]
-	if restartInfo == nil {
+	restartInfo := r.historyByService[t.ServiceID][instanceTuple]
```

Comment:
Perhaps I'm mis-interpreting the TODO at the top;

```
// TODO(aluzzardi): This function should not depend on `service`.
```

But doesn't this make it even more dependent on `service`? (I agree it's cleaner :smile:)